andela/jubilee-backend
# #167733642 Build out REST Endpoint for signup

### What does this PR do?
Have user registration endpoint with token generation on successful signup
### Description of Task to be completed?
- Install dependencies
- Verify app connects to the database
- Create a User model
- Create a corresponding User migration file
- Set up scripts for running migration commands
- Refactor app to use JWT instead of session
- Create a User service that interface with the database
- Create a User controller that saves the user in the database
- Create signup route
- Write the test for signup route

### How should this be manually tested?
- Clone this repository
- Copy environment variable sample from .env.sample into .env and fill in the required variables
- On the root directory, run docker-compose up -d to start up both app and a database instance on docker
- Alternatively, run npm start-dev on project root directory to bypass docker. but ensure to provide the right database connections
- Run npm run migrate to run migration file. Check your database for a newly added Users table
- Run npm run migrate:reset to drop said Users table
- On Postman, call `POST /api/auth/signup` with
```
{
    firstName: string,
    lastName: string,
    email: string,
    gender: string,
    street: string;
    city: string,
    state: string,
    country: string,
    bithdate: Date,
    phoneNumber: string,
    companyName: string
}
```
### Any background context you want to provide?
As mentioned above, this feature can be tested with or without docker. This is also a prerequisite for a lot of other features hence it needs to be reviewed in order to unblock team members
### What are the relevant pivotal tracker stories?
[#167733642](https://www.pivotaltracker.com/story/show/167733642)
